File: PP1/pages/api/code/temp/main.ts
// Create a function to read input using process.stdin
function getInput(): Promise<string> {
  return new Promise((resolve) => {
    process.stdout.write("Enter something: ");
    
    process.stdin.once("data", (data) => {
      resolve(data.toString().trim());  // Convert input buffer to string
      process.stdin.pause();  // End the input stream
    });
  });
}

// Usage of getInput function
getInput().then((userInput) => {
  console.log(`You entered: ${userInput}`);
});

File: PP1/pages/api/code/temp/Main.rs
use std::io;

fn main() {
    println!("Enter something:");

    // Create a mutable String to hold the input
    let mut input = String::new();

    // Read input from the standard input
    match io::stdin().read_line(&mut input) {
        Ok(_) => {
            println!("You entered: {}", input.trim());
        }
        Err(e) => {
            println!("Failed to read input: {}", e);
        }
    }
}
